### PR TITLE
nested phi optimization changes

### PR DESCRIPTION
**Pull request by JohnTortugo**

### CODE REVIEW

**Comment by JohnTortugo** [comment]
@dhanalla - Please update your branch with the changes to the master branch.

File: src/hotspot/share/opto/escape.cpp [hunk 1]
```diff
@@ -460,51 +460,66 @@ bool ConnectionGraph::can_reduce_phi_check_inputs(PhiNode* ophi) const {
...
+         NOT_PRODUCT(if (TraceReduceAllocationMerges) tty->print_cr("Can NOT reduce Phi %d on invocation %d. AddP user isn't a [splittable] Load(): %s", ophi->_idx, _invocation, use_use->Name());)
+         return false;
         }
       }
+    } else if (nestedDepth > 1) {
+      // Note: Do not change the position of this else as all below cases are not eligible for nested optimizations
+      return false;
+    } else if (use->is_Phi()) {
+      if (ophi->_idx == use->_idx) {
+	tty->print_cr("***Can reduce self loop nested phi***");
```
Comment (JohnTortugo) [inline src/hotspot/share/opto/escape.cpp:498]: Does this really work? In any case, you need to check the debug flags before printing this message.

**Review by JohnTortugo — COMMENTED** [review]
Just a few comments for now.

Please take a look at OpenJDK code style as there are some places in your code that don't follow the guideline.

**Comment by dhanalla** [comment]
close the old PR.

**Pull request closed without merging**